### PR TITLE
evsieve: init at 1.3.1

### PR DESCRIPTION
**Pull request by tsowell**
###### Description of changes

Add evsieve: "A utility for mapping events from Linux event devices"
https://github.com/KarsMulder/evsieve

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by solson — APPROVED** [review]
Looks good to me. I've been using `evsieve` daily for a year and my personal derivation is almost identical to this one.

File: pkgs/tools/inputmethods/evsieve/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ pkgs, lib, fetchFromGitHub, rustPlatform, ... }:
```
Comment (pbsds) [inline pkgs/tools/inputmethods/evsieve/default.nix:1]:
```suggestion
{ lib
, fetchFromGitHub
, rustPlatform
, libevdev
}:
```

File: pkgs/tools/inputmethods/evsieve/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    owner = "KarsMulder";
+    repo = "evsieve";
+    rev = "v${version}";
+    sha256 = "sha256-R/y3iyKGE4dzAyNnDwrMCr8JFshYJwNcgHQ8UbtuRj8=";
+  };
+
+  cargoSha256 = "sha256-jkm+mAHejCBZFalUbJNaIxtIl2kwnlPR2wsaYlcfSz8=";
+
+  buildInputs = [ pkgs.libevdev ];
```
Comment (pbsds) [inline pkgs/tools/inputmethods/evsieve/default.nix:16]:
```suggestion
  buildInputs = [ libevdev ];
```

File: pkgs/tools/inputmethods/evsieve/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "KarsMulder";
+    repo = "evsieve";
+    rev = "v${version}";
+    sha256 = "sha256-R/y3iyKGE4dzAyNnDwrMCr8JFshYJwNcgHQ8UbtuRj8=";
+  };
+
+  cargoSha256 = "sha256-jkm+mAHejCBZFalUbJNaIxtIl2kwnlPR2wsaYlcfSz8=";
```
Comment (pbsds) [inline pkgs/tools/inputmethods/evsieve/default.nix:14]:
```suggestion
  cargoHash = "sha256-jkm+mAHejCBZFalUbJNaIxtIl2kwnlPR2wsaYlcfSz8=";
```

File: pkgs/tools/inputmethods/evsieve/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "evsieve";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "KarsMulder";
+    repo = "evsieve";
+    rev = "v${version}";
+    sha256 = "sha256-R/y3iyKGE4dzAyNnDwrMCr8JFshYJwNcgHQ8UbtuRj8=";
```
Comment (pbsds) [inline pkgs/tools/inputmethods/evsieve/default.nix:11]:
```suggestion
    hash = "sha256-R/y3iyKGE4dzAyNnDwrMCr8JFshYJwNcgHQ8UbtuRj8=";
```

**Review by pbsds — CHANGES_REQUESTED** [review]
Looking good! 

Please avoid using `pkgs` in nixpkgs derivations. It's nice while testing, but it hampers overridability.

**Comment by tsowell** [comment]
Thank you reviewing this, and thanks for the advice!  I've made the suggested changes (and I also rebased).

**Review by pbsds — APPROVED** [review]

**Pull request merged**